### PR TITLE
Adding using reduce

### PR DESCRIPTION
**Pull request by OGordoo**

### CODE REVIEW

File: exercise/reduce.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+## Reduce
```
Comment (kigiri) [inline exercise/reduce.js]: Confusing name, why is it called reduce ?
Also, what has this exercise to do with callbacks ?

**Review by kigiri — CHANGES_REQUESTED** [review]

File: exercise/reduce.js (hunk 1 above)
Comment (OGordoo) [reply exercise/reduce.js]: is for the students have to use reduce, i can think of some other way for them to use it

File: exercise/reduce.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+Create a function called `sumOrMul` that receives an array and adds or
+multiplies its elements depending on whether the element is an odd or an even number.
+
+If the element is even the function should multiply it,
+otherwise it should add it.
+
+The function returns the final value.
+
+Example:
+sumOrMul([1, 2, 3, 4]) = ((1 * 2) + 3) * 4 = 20
```
Comment (kigiri) [inline exercise/reduce.js]: code should be in \`backticks\` also, since single `=` is used to assign values.
```js
sumOrMul([1, 2, 3, 4])
	// -> ((1 * 2) + 3) * 4
    // -> 20
```


As for using reduce, I feel like we should include the good old classic `total`
```js
[1,2,3].reduce((a,b) => a + b, 0)
```
We use it all the time it's juste the simplest usecase for reduce.
And also do tests with empty arrays to be `0`.


Another good one is applying an array of functions:
```js
[ x => x + 2, x => x ** 2, x => x - 7, x => `result: [${x}]` ]
	.reduce((x, fn) => fn(x), 10)
	// -> 'result: [137]'
```

We want to keep the function passed to reduce to the minimum to demonstrate what reduce do.
The exercise must not be centered arround conditions.

**Review by kigiri — APPROVED** [review]
